File: 백준/Gold/1806. 부분합/부분합.py
N, S = map(int, input().split())
arr = list(map(int, input().split()))
left, right, cur_sum, ans = 0, 0, 0, 0x7fff_ffff

while left <= right:
    if cur_sum >= S:
        ans = min(ans, right - left)
        cur_sum -= arr[left]
        left += 1
    elif right == N: break
    else:
        cur_sum += arr[right]
        right += 1

if ans == 0x7fff_ffff:
    print(0)
else:
    print(ans)

File: boj/TwoPointer_BinarySearch/부분합.cpp
#include <iostream>
using namespace std;

int main(void) {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int n, s;
    int arr[100000];
    int ans = 0x7fffffff;

    cin >> n >> s;
    for (int i = 0; i < n; i++) cin >> arr[i];

    int l = 0, r = 0, acc = arr[0];
    while (l <= r && r < n) {
        if (acc < s) {
            r++;
            if (r >= n) break;
            acc += arr[r];
        }
        else {
            ans = min(ans, r - l + 1);
            acc -= arr[l++];
        }
    }

    if (ans == 0x7fffffff)
        cout << 0;
    else
        cout << ans;
    return 0;
}